core/editor/src/tools/mod.rs

```diff
@@ -20,7 +38,23 @@ impl Default for ToolState {
 			trace: Trace::new(),
 			primary_color: Color::BLACK,
 			secondary_color: Color::WHITE,
-			active_tool: ToolType::Select,
+			active_tool_type: ToolType::Select,
+			tools: {
+				let mut hash_map: HashMap<ToolType, Box<dyn Tool>> = HashMap::new();
+
+				hash_map.insert(ToolType::Select, Box::new(select::Select::default()));
+				hash_map.insert(ToolType::Crop, Box::new(crop::Crop::default()));
+				hash_map.insert(ToolType::Navigate, Box::new(navigate::Navigate::default()));
+				hash_map.insert(ToolType::Sample, Box::new(sample::Sample::default()));
+				hash_map.insert(ToolType::Path, Box::new(path::Path::default()));
+				hash_map.insert(ToolType::Pen, Box::new(pen::Pen::default()));
+				hash_map.insert(ToolType::Line, Box::new(line::Line::default()));
+				hash_map.insert(ToolType::Rectangle, Box::new(rectangle::Rectangle::default()));
+				hash_map.insert(ToolType::Ellipse, Box::new(ellipse::Ellipse::default()));
+				hash_map.insert(ToolType::Shape, Box::new(shape::Shape::default()));
```

Comment:
maybe use the `maplit` crate (a small dependency) to make this code easier to maintain?

Reply:
That would be pretty nice, should I make a similar macro to the one from `maplit` or should I just use the crate?

Reply:
I think using a similar macro would be better since

1. It doesn't add a dependency
2. We can also include `Box::new` and `ToolType::` in the macro, so we only have to provide the enum variants when invoking the macro

be sure to initialize the hashmap with `HashMap::with_capacity` though